src/hotspot/cpu/ppc/icache_ppc.hpp

```diff
@@ -44,7 +44,7 @@ class ICache : public AbstractICache {
   static void ppc64_flush_icache_bytes(address start, int bytes) {
     // Align start address to an icache line boundary and transform
     // nbytes to an icache line count.
-    const uint line_offset = mask_address_bits(start, line_size - 1);
+    const uint line_offset = (intptr_t)start & (line_size - 1);
```

Comment:
Drive by comment: This used to be a `uintptr_t` cast, now its `intptr_t`- why signed? Was that a deliberate decision?

Side question, I always wondered about the widespread use of `intptr_t` in hotspot when casting pointers to integrals, e.g. in `p2i()`. Why a signed integer? E.g. if you then use it for bitwise ops (the major reason why one converts a pointer to an int) it seems more hassle to have to remember the rules for using bitwise operators on signed integers.

Reply:
Hi Thomas, thanks for noticing.  I messed this one up.

Are you arguing for address_word to avoid this confusion?  That's a lot of reeducation in the code.

Reply:
Hi Coleen, no, I was just wondering if the decision was deliberate. I may have overlooked a reason for this.

Reply:
@tstuefe  `p2i()` is mainly used for printing, and the format uses hex so the signed-ness is immaterial. But in general, yes this seems an odd choice.